elastalert/util.py

```diff
@@ -60,27 +61,46 @@ def _find_es_dict_by_key(lookup_dict, term):
     # For example:
     #  {'foo.bar': {'bar': 'ray'}} to look up foo.bar will return {'bar': 'ray'}, not 'ray'
     dict_cursor = lookup_dict
-    subkeys = term.split('.')
-    subkey = ''
 
-    while len(subkeys) > 0:
-        if not dict_cursor:
-            return {}, None
+    while term:
+        split_results = re.split(r'\[(\d?)\]', term, 1)
```

Comment:
Maybe for readability you can use the keyword for `maxsplit`, `split_results = re.split(r'\[(\d?)\]', term, maxsplit=1)`